### PR TITLE
feat: Add compose to android app

### PR DESCRIPTION
**Pull request by Balcan**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue: minSDK to 21 ](https://jira.dhis2.org/browse/ANDROAPP-4033)
[ jira issue: Java 11 support ](https://jira.dhis2.org/browse/ANDROAPP-4304)
[ jira issue:  bump Kotlin version](https://jira.dhis2.org/browse/ANDROAPP-4013)
[ jira issue: bump gradle versions ](https://jira.dhis2.org/browse/ANDROAPP-3923)

**NOTE**
CI passed. It needed the java 8 step to be removed (I've added a new primary-j11 workflow):
https://app.bitrise.io/build/c4a7975f-44fb-4bf8-a5d6-234ce3f00104#?tab=log

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -55,12 +54,14 @@ android {
...
+        buildConfigField("long", "VERSION_CODE", "${defaultConfig.versionCode}")
+        buildConfigField("String","VERSION_NAME","\"${defaultConfig.versionName}\"")
 
-        javaCompileOptions {
+        /*javaCompileOptions {
             annotationProcessorOptions {
                 includeCompileClasspath true
             }
-        }
+        }*/
```
Comment (ferdyrod) [inline app/build.gradle:64]: Commented code

File: app/src/main/java/org/dhis2/usescases/development/DevelopmentActivity.java [hunk 2]
```diff
@@ -57,9 +58,11 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 
     private void loadProgramRuleCheck() {
         binding.ruleActionQualityButton.setOnClickListener(view -> {
-            binding.ruleActionQualityResult.setText("Checking...");
+            Intent intent = new Intent(this,ProgramRulesValidationActivity.class);
+            startActivity(intent);
+            /*binding.ruleActionQualityResult.setText("Checking...");
             String result = new ProgramRulesValidations(D2Manager.getD2()).validateRules();
-            binding.ruleActionQualityResult.setText(result);
+            binding.ruleActionQualityResult.setText(result);*/
```
Comment (ferdyrod) [inline app/src/main/java/org/dhis2/usescases/development/DevelopmentActivity.java:65]: Commented code

File: app/src/main/java/org/dhis2/usescases/development/ProgramRulesValidationsModule.kt [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import org.dhis2.form.ui.style.FormUiColorFactory
+import org.dhis2.form.ui.validation.FieldErrorMessageProvider
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.D2Manager
+
+@Module
+class ProgramRulesValidationsModule(val context: ProgramRulesValidationActivity) {
+
+    @Provides
+    @PerActivity
```
Comment (ferdyrod) [inline app/src/main/java/org/dhis2/usescases/development/ProgramRulesValidationsModule.kt:26]: I could be wrong here, but I thought that with the new version of dagger, there is no need to specify this type of annotation.

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.1.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.0.2-all.zip
```
Comment (ferdyrod) [inline gradle/wrapper/gradle-wrapper.properties:6]: In the build.gradle file you are using version 7.0.3 of the gradle plugin, maybe we can use the same in here.

**Review by ferdyrod — COMMENTED** [review]
Since all usecases dagger module files are going to be modified, maybe is a goo time too remove the _created by_ comment on top of each file.

Great job!!!

File: app/src/main/java/org/dhis2/usescases/development/ProgramRulesValidationsModule.kt (hunk 3 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/development/ProgramRulesValidationsModule.kt:26]: They are not needed after the @Module tag. We can check, but that would be part of other ticket

File: gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (Balcan) [reply gradle/wrapper/gradle-wrapper.properties:6]: Funny thing: 7.0.3-all does not exist, build fails. So for now we will keep as it is.

**Comment by Balcan** [comment]
> Since all usecases dagger module files are going to be modified, maybe is a goo time too remove the _created by_ comment on top of each file.
> 
> Great job!!!

I'm going to create a ticket to remove the "created by" from all classes

File: app/build.gradle (hunk 1 above)
Comment (Balcan) [reply app/build.gradle]: done

File: app/src/main/java/org/dhis2/usescases/development/DevelopmentActivity.java (hunk 2 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/development/DevelopmentActivity.java]: done

**Review by andresmr — APPROVED** [review]

**Pull request merged**